tests/Engine.Tests/GameEngineTests.cs

```diff
@@ -42,6 +42,38 @@ public void WhenConfigDoesNotProvideOptionsThenThrowsException()
             }
         }
 
+        public class Dispose
+        {
+            [UnitTest]
+            public void WhenDisposedThenShouldThrowObjectDisposedExceptionWhenPublish()
+            {
+                // TODO(DANjEEEEE): I'm not sure if this is the correct way to test / implement this, so any advice would be useful.
+                var engine = new GameEngine<Fake.GameTime>(DefaultConfig, DefaultTranslator, DefaultClock);
+                using (engine) { }
```

Comment:
Interesting. Personally I would just call `engine.Dispose()` instead of the `using`.